src/lib/operators/table_scan.cpp

```diff
@@ -113,7 +113,8 @@ std::shared_ptr<const Table> TableScan::_on_execute() {
       if (_in_table->get_type() == TableType::References) {
         const auto chunk_in = _in_table->get_chunk(chunk_id);
 
-        auto filtered_pos_lists = std::map<std::shared_ptr<const PosList>, std::shared_ptr<PosList>>{};
+        auto filtered_pos_lists =
+            std::map<std::shared_ptr<const PosList>, std::pair<std::shared_ptr<PosList>, PosListType>>{};
```

Comment:
When would we want a PosList without that type information? Does it make sense to incorporate that in `PosList`?

Reply:
Would be nice, but how would you do that? PosList is just an alias for an `std::vector` and it’s considered bad practice to inherit from container types, among others because they don’t have virtual destructors. So the other option would be to create a new class and duplicate the interface.

Reply:
Unfortunately, I don't know of any better way

Reply:
Or we create a simple struct:

```c++
struct PosList {
  std::vector<RowID> positions;
  PosListType type;
};
```

Reply:
In 90% of the cases, we are interested in `positions`. Adding the indirection there seems like a bad choice.